projects/plugins/boost/app/modules/cache/Page_Cache.php

```diff
@@ -80,15 +80,20 @@ private function create_advanced_cache() {
 
 		if ( file_exists( $advanced_cache_filename ) ) {
 			$content = file_get_contents( $advanced_cache_filename ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
-			if ( strpos( $content, Page_cache::$advanced_cache_signature ) !== false ) {
+			if ( strpos( $content, self::$advanced_cache_signature ) !== false ) {
 				return true;
 			} else {
 				return new \WP_Error( 'advanced-cache.php exists but is not the correct file' );
 			}
 		} else {
-			$contents = '<?php
-// ' . Page_cache::$advanced_cache_signature . '
-require_once( ABSPATH . \'/wp-content/plugins/boost/app/modules/cache/Boost_File_Cache.php\' );
+			$plugin_name = basename( dirname( plugin_dir_path( __FILE__ ), 3 ) );
+			$filename    = WP_CONTENT_DIR . '/plugins/' . $plugin_name . '/app/modules/cache/Boost_File_Cache.php';
```

Comment:
```suggestion
			$filename = dirname( plugin_dir_path( __FILE__ ), 3 ) . '/app/modules/cache/Boost_File_Cache.php';
```

Why go through all the complication of unpacking the plugin name _from its absolute path_ only to stitch it back together in a way that may break?

Reply:
While we're at it, calling the path to `Boost_File_Cache` `$filename` in a function that also deals with other kinds of filename is confusing. How about `$boost_cache_filename` ? :)

Reply:
If I don't use basename() to get the plugin name, it may use a path outside of the htdocs or public_html directory if Boost is symlinked from somewhere else. I think that's the issue that was the cause for me to add the WPCACHEHOME constant for WPSC. I'll leave basename() as is.
I will rename the variable. I was going to use that variable name initially, but it seemed so long I shortened it. :)

Reply:
I've just spent the past 20 minutes going down the rabbit hole of WordPress directory management. 🐰 🕳️ 😱 

WordPress [discourages the use of constants like `WP_CONTENT_DIR` in plugins](https://codex.wordpress.org/Determining_Plugin_and_Content_Directories#Constants) and asks that we use functions like `plugin_dir_path` (which is a thin wrapper around `dirname`) instead... But we use it all over Jetpack. There doesn't seem to be a cleaner way to ask where the content lives.

But the docs also caution about weird behaviour around symlinks too, as you are worried about here... But I'm not convinced we'll hit such an issue if we're not stepping out of the Boost directory with dirname().

All up? I'm not sure what the best pattern here is.

In the spirit of keeping this project moving forward, I'll go with your instincts on this one because you have so much experience here. We do check the file exists before we require it, so if we get it wrong in some weird edge-cases, the worst that will happen is the cache system won't run.

Given that I'll 👍 this. But I'm still leery of how WordPress handles paths around this.